### PR TITLE
Update benchmarks for Python 3.12

### PR DESCRIPTION
**Pull request by movermeyer**
### What are you trying to accomplish?

Updating the benchmarking metrics in the tables. 

### What approach did you choose and why?

Ran the benchmarking scripts.

I had to exclude `pendulum` and `maya` (which uses `pendulum`) since `pendulum` won't build on Python 3.12 until v3.0.0 comes out (any day now???): https://github.com/sdispater/pendulum/issues/696

`tox` no longer supports Python < 3.7, so it isn't useful for that anymore.
This means that the Python 2.7 data hasn't been updated. 😞

### What should reviewers focus on?

There should be no changes to the Python 2.7 column. I had to manually insert this data, so that's the spot that's most likely to have broken, though I've double-checked.

### The impact of these changes

Users can see our Python 3.12 performance numbers and feel that we are keeping up with modern Python.

### CODE REVIEW

File: benchmarking/format_results.py [hunk 1]
```diff
@@ -128,8 +128,8 @@ def write_benchmarking_results(results_directory, output_file, baseline_module,
 
     writer = pytablewriter.RstGridTableWriter()
     formatted_python_versions = [f"Python {major}.{minor}" for major, minor in python_versions_by_modernity]
-    writer.header_list = ["Module"] + (["Call"] if include_call else []) + formatted_python_versions + [f"Relative slowdown (versus {baseline_module}, latest Python)"]
-    writer.type_hint_list = [pytablewriter.String] * len(writer.header_list)
+    writer.headers = ["Module"] + (["Call"] if include_call else []) + formatted_python_versions + [f"Relative slowdown (versus {baseline_module}, latest Python)"]
+    writer.type_hints = [pytablewriter.String] * len(writer.headers)
```
Comment (movermeyer) [inline benchmarking/format_results.py:132]: `pytablewriter` v1.0.0 dropped these, so I replaced them with the new names: https://github.com/thombashi/pytablewriter/commit/77770338b7ffdefa1aa8c45b0de1137d529a9fcd

**Review by AlecRosenbaum — APPROVED** [review]
Not a huge deal, but on the second table the extra column makes the table wider than the page will go (and looks cut-off):

image

The relative speeds are probably most important to at a glance, so maybe those should be the first column before the python versions? (Absolutely not a strong opinion here, just figured I'd mention it)

**Comment by movermeyer** [comment]
> Not a huge deal, but on the second table the extra column makes the table wider than the page will go (and looks cut-off):

@AlecRosenbaum Very true. I've switched the order of the columns so that the relative speed-up is the second column.

----

Aside: for my future reference:

As for making the columns thinner, I thought about changing `Incorrect Result` to `❌` and having a tooltipto explain what the symbol meant. The only way that I found to get GitHub to render it was using the `..raw: html` block in the cell:

```rst
.. table::

  +--------------------------------------------------------+
  |             Result                                     |
  +========================================================+
  |.. raw:: html                                           |
  |                                                        |
  |   <span title="Incorrect Result (``None``)">❌</span>  |
  +--------------------------------------------------------+
```

[But `pytablewriter` doesn't support multi-line values in table cells.](https://github.com/thombashi/pytablewriter/issues/32). 🤷

**Comment by movermeyer** [comment]
Alright, instead of changing the column order (I was partial to what we had), I've done the simple change of changing the `Incorrect Result` to `❌`, without a tooltip. The columns fit once again:

image

File: benchmarking/format_results.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def __str__(self):
         if self.exception:
             return f"Raised  ``{self.exception}`` Exception"
         elif not self.matched_expected:
-            return f"**Incorrect Result** (``{self.parsed_value}``)"
+            return "❌"
```
Comment (movermeyer) [inline benchmarking/format_results.py:26]: Reduced width so that the columns all fit without a scroll bar.

**Comment by movermeyer** [comment]
But of course, the table width shown in PR review is not the same as shown on the [project main page](https://github.com/closeio/ciso8601) 🤦:

image

so the column re-ordering will be necessary (unless we drop some columns):

**Pull request merged**